### DISCUSSION TITLE
Relations between x y axis and the velocity in pixels?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1093] [Q&A] [answered]
Hi:

We tried to compare the two velocities and find the relationship between x, y coordinates and the velocity in pixels
the 1st is simply as 
```java
var velocity = 6;
physics.setVelocityX(FXGL.getPhysicsWorld().toPixels(velocity);
```
the 2nd is
```java
var velocity = 6;
entity.setX(entity.getX + velocity);
```
in theory they should be equal right?
but seems like they are slight different
the 2nd has larger velocity
so if we set 2nd velocity to 5
then seems both are equal
not very sure it is because of this line of source code
```java
jboxWorld.setParticleDensity(1.2f);
```
so any idea why this happened?
And how to fix this descrepency?

Thanks

**Comment by AlmasB** [comment, marked as answer]
There are two methods in PhysicsComponent that take velocity: one is in pixels, the other in meters. Velocity is measured by units per second, rather than units per frame (at least as far as I can remember), so something like `x += velocity * tpf` is a closer comparison.

**Reply by chengenzhao** [reply]
Got it
```java
physics.setVelocityX(500);
```
==
```java
entity.setX(entity.getX() + 500*tpf);
```
